Removes lack of mutant plants

TobyThorne opened the pull request:
Огромная куча всякой всячины для увеселения жизни ботаников.

**Добавлены/изменены следующие растения:**
* Томат-убийца (killer tomato). Старый-добрый томат убийца, исчезнувший из-за смешных переименований растений. В итоге вместо "tomatokiller" в мутациях стояло "killer" и нихрена не работало.
Мутирует из: bloody tomato
Мутирует в: N/A
* Томак (Tomacco). Съедобные сигареты! 
Мутирует из: tomato
Мутирует в: N/A
* Нанокровавый Томат (Nanoblood Tomato). Как кровавый томат, только с нанокровью.
Мутирует из: blood tomato
Мутирует в: automato
* АвТомат (Automato). Содержит наниты! Пик томатной селекции.
Мутирует из: nanoblood tomato
Мутирует в: N/A
* Баклажан (Eggplant). Возвращена старая мутация.
Мутирует из: N/A
Мутирует в: egg plant
* Яичное растение (Egg plant). Несёт цветные яйца. Почему нет?
Мутирует из: eggplant
Мутирует в: N/A
* Красные яблоки (Red apples). Бывшие обычные яблоки. Содержит яблочный сок.
Мутирует из: yellow apples, green apples
Мутирует в: yellow apples, green apples, poison apples
* Жёлтые яблоки (Yellow apples). Тоже содержат яблочный сок.
Мутирует из: red apples, green apples
Мутирует в: red apples, green apples, golden apples
* Зелёные яблоки (Green apples). Как ни странно, сок.
Мутирует из: red apples, yellow apples
Мутирует в: red apples, yellow apples, radioactive apples
* Ядовитые яблоки (Poison apples). Не отличаются от обычных красных яблок, содержат цианид.
Мутирует из: red apples
Мутирует в: N/A
* Золотые яблоки (Golden apples). Содержат золото. Блестят.
Мутирует из: yellow apples
Мутирует в: N/A
* Радиоактивные яблоки (Radioactive apples). Содержат радий. Избавляют от необходимости постоянно бегать к химикам.
Мутирует из: green apples
Мутирует в: N/A
* Подлунник (Moonflower). Содержит самогон.
Мутирует из: sunflower
Мутитрует в: N/A
* Банан (Banana). Добавлены мутации.
Мутирует из: N/A
Мутирует в: lubanana, bunana
* Бунан (Bunana). Большой, круглый банан. Похож на булочку.
Мутирует из: banana
Мутирует в: N/A
* Любанан (Lubanana). ОЧЕНЬ скользкий и содержит смазку. Мечта клоуна. Или ЕРПшника. 
Мутирует из: banana
Мутирует в: BANana
* БАНан (BANana). При применении обсепечивает мгновенный сердечный приступ всем игрокам в зоне видимости.
Мутирует из: lubanana
Мутирует в: N/A
* Картоха (Potato). Добавлена мутация.
Мутирует из: N/A
Мутирует в: electric potato
* Электрокартоха (Electric potato). Производит готовые картофельные батарейки.
Мутирует из: potato
Мутирует в: N/A
* Кокос (Coconut). Источник кокосового молока. Вкусно.
Мутирует из: N/A
Мутирует в: cowcownut
* Кокорова (Cowcownut). Самая ужасная игра слов, на которую меня хватило. Кокосовую коровку, внезапно, можно доить ради кокосового молока.
* Диона (Diona). Добавлена лучшая мутация.
Мутирует из: N/A
Мутирует в: maneater
* Людоед (Maneater). Зубастая ебака, которую сложно получить из-за склонности растущей дионы умирать от радиации и сложно размножить из-за отсутствия семян. Зато можно создать личную цветочную гвардию и дать по сосать всем в радиусе одной станции. После сбора с грядки в течение пяти секунд приходит в себя, в это время можно скормить людоеду кусок мяса - покормившего космонавтика оно атаковать не будет. Дружит с томатами-убийцами.
Мутирует из: diona
Мутирует в: N/A
* Чили (chili). Добавлена мутация.
Мутирует из: N/A
Мутирует в: cold pepper, flame chili
* Огненный чили (Flame chili). Содержит топливо. Если добиться взрывного гена - будет смешно.
Мутирует из: chili
Мутирует в: N/A

**Следующих мобов теперь можно перехватывать, играя призраком:**
- Ходячие грибочки (Walking Mushrooms). Безвредны, но могут распространять свои споры. И вообще милашки. 
- Томаты-убийцы (Killer Tomatoes). Не слишком опасные, зато милые. Джвесять лет ждал возможности поотыгрывать помидор.

**Прочие изменения:**
- Наниты возвращены из состояния кастрированного плейсхолдера. В маленьких дозах действуют как очень мощная версия трика, залечивая ожоги и брут. При передозе (5+ единиц) с некоторым шансом превращают космонавтика в борга, раскидывая слабую плоть повсюду в радиусе некоторых метров. Как это и было раньше, в общем-то.
- Большая часть спрайтов растений заменена на гуновские, охрененно красивые.
- Пофикшена куча мелких недочётов у растений: отвалившийся цвет у зелёного винограда, спрайт-плейсхолдер у чеснока и лука, etc.
- Добавлены кокосовое молоко (надо будет потом добавить рецепты вроде пина колады, но сейчас лень) и яблочный сок (немножко лечебный, с рецептами то же самое).
- Реагенты из сочных фруктов при кидании (и последующем размазывании) теперь переносятся на всё, что находится на тайле. Например, любананы мажут пол смазкой, крапива с геном сочности сносит лица космонавтикам, огненный чили разливает горючку.

- [X] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [X] Я внимательно прочитал все свои изменения и багов в них не нашел. <-- Врёт <-- Не врёт
- [X] Я запускал сервер со своими изменениями локально и все протестировал. <-- тож врёт инфа сотка <-- Не вру риии это гит багов наплодил
- [X] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

Commit:
Hyper Hype Hydroponics Hupdate

Added plants:
- Killer tomatoes
- Tomacco
- Nanoblood Tomato
- Automato
- Real Eggs Plant
- Red apples
- Yellow apples
- Green apples
- Radioactive apples
- Moonflowers
- Lubananas
- Bunanas
- BANanas
- Electric potatoes
- Coconuts
- Cowcownuts
- Maneaters

The following mobs are now possessable:
- Walking Mushrooms
- Killer Tomatoes
- Maneaters

- Nanites can now be used as a powerful medicine, but may gib one into a cyborg upen overdosing
- Replaced many plants' sprites by Goonstation ones
- Added apple juice and coconut milk;

Pos13 commented:
HYPE

TapKo4eB commented on code/game/objects/items/toys.dm:
```diff
@@ -937,3 +938,21 @@
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "banana"
+
+/obj/item/toy/banbanana/attack_self(mob/user as mob)
```
as mob не нужен же

TobyThorne replied:
Скопипастил кокойто легаси. Упс.

TapKo4eB commented on code/modules/hydroponics/seed_storage.dm:
```diff
@@ -50,53 +50,56 @@
+		/obj/item/seeds/towermycelium 			= 30,
+		/obj/item/seeds/watermelonseed 			= 30,
+		/obj/item/seeds/wheatseed 				= 30,
+		/obj/item/seeds/whitebeetseed 			= 30
```
![image](https://user-images.githubusercontent.com/48761132/81493244-882bff80-92a7-11ea-819e-61d3167137e7.png)
![image](https://user-images.githubusercontent.com/48761132/81493233-79454d00-92a7-11ea-9ffc-0fba24874210.png)

[выранивание табами аааа](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md#%D1%82%D0%B0%D0%B1%D1%8B-vs-%D0%BF%D1%80%D0%BE%D0%B1%D0%B5%D0%BB%D1%8B)

TobyThorne replied:
блет гит хуб за што :(

TobyThorne committed:
tabs suck

TobyThorne committed:
commit/C as legacy|shit

TapKo4eB commented on code/modules/mob/living/simple_animal/hostile/maneater.dm:
```diff
@@ -0,0 +1,69 @@
+		angry = 1
+		faction = "creature"
+
+/mob/living/simple_animal/hostile/maneater/attackby(obj/item/O as obj, mob/user as mob)
```
ищё тут

TapKo4eB commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -129,7 +132,7 @@
-			udder.add_reagent(/datum/reagent/drink/milk, rand(5, 10))
+			udder.add_reagent(milktype, rand(5, 10))
 
 /mob/living/simple_animal/cow/attack_hand(mob/living/carbon/M as mob)
```
тут

TapKo4eB commented on code/modules/mob/living/simple_animal/friendly/farm_animals.dm:
```diff
@@ -106,18 +106,21 @@
+
 	..()
 
 /mob/living/simple_animal/cow/attackby(obj/item/O as obj, mob/user as mob)
```
и тут

Commit:
u+b+ly+at

Commit:
Merge branch 'hydroponchik' of https://github.com/TobyThorne/OnyxBay into hydroponchik

TapKo4eB commented:
![image](https://user-images.githubusercontent.com/48761132/81495840-e367ed00-92bb-11ea-9403-2ac939c7529f.png)
я надеюсь пакет семян бунаны так и должен выглядеть?

TapKo4eB commented on code/modules/hydroponics/seed_datums.dm:
```diff
@@ -708,13 +796,49 @@
+	seed_name = "lubanana"
+	display_name = "lubanana tree"
+	chems = list(/datum/reagent/lube = list(10,10))
+	mutants = ("banbanana")
```
```suggestion
	mutants = list("banbanana")
```
э сюка сломал блеать прок `get_mutant_variant()` вот именно банбанане чёртовы кодеры кодят а потом люди фичей воспользоваться не смогут

TobyThorne replied:
не баг а фича инфа сотка 💯💯💯

TobyThorne committed:
Update code/modules/hydroponics/seed_datums.dm

Co-authored-by: TapKo4eB <48761132+TapKo4eB@users.noreply.github.com>

ArmenianOnyx commented:
Скорее принимайте, жирный плюсик.

dnivji commented:
Я геймдиз простой, вижу ПР Тоби, ставлю аппрув.

Commit:
Hypdate

- Fixed bunana seed icon
- Added flame chilis
- Lubananas are EXTRA SLIPPERY now
- Throwing juicy fruits now transfers their reagents on the target tile

TobyThorne commented:
Добавил перенос реагентов при размазывании фрукта. И любананы теперь ещё и сами по себе скользкие.

TapKo4eB commented on code/modules/hydroponics/seed_datums.dm:
```diff
@@ -708,13 +808,49 @@
+	seed_name = "lubanana"
+	display_name = "lubanana tree"
+	chems = list(/datum/reagent/lube = list(10,10))
+	mutants = ("banbanana")
```
![image](https://user-images.githubusercontent.com/48761132/81503812-69049080-92ee-11ea-9327-3bf0bb2b7d78.png)
зачем ты добавляешь в код то что нельзя будет получить игровым путём

TobyThorne replied:
бля нахуй код ночью писать, тупо говна попил %(

TapKo4eB reviewed (changes requested):
блять я аппрувнул

TobyThorne committed:
check description

00111111 00111110 00111111 00111000 00111011 00100000 00110011 00111110 00110010 00111101 00110000

Moscowchief commented:
Аппрув!

Epicus7 commented:
Да это же платный иссуй #1216

TapKo4eB reviewed (dismissed):
реальн кстати платный

Epicus7 commented on code/modules/mob/living/simple_animal/hostile/maneater.dm:
```diff
@@ -0,0 +1,69 @@
+
+	var/angry = 0
+
+/mob/living/simple_animal/hostile/maneater/Initialize()
+	. = ..()
+
+	if(name == initial(name))
+		name = "[name] ([sequential_id(/mob/living/simple_animal/hostile/maneater)])"
+	real_name = name
+
+	spawn(5 SECONDS)
+		angry = 1
+		faction = "creature"
+
+/mob/living/simple_animal/hostile/maneater/attackby(obj/item/O, mob/user)
+	if(!angry && O.icon_state == "meat")
+		if(!stat)
+			for(var/mob/M in viewers(user, null))
+				if ((M.client && !( M.blinded )))
+					M.show_message("<span class='notice'>[user] feeds the [name] with some [O].</span>")
+			user.drop_from_inventory(O)
+			qdel(O)
+			angry = 1
+			friends += user
+	else
+		..()
+
+/mob/living/simple_animal/hostile/maneater/FindTarget()
+	. = ..()
+	if(.)
+		playsound(get_turf(src), 'sound/voice/MEraaargh.ogg', 70, 1)
+		custom_emote(1,"roars at [.]")
+
+/mob/living/simple_animal/hostile/maneater/AttackingTarget()
+	. =..()
+	var/mob/living/L = .
+	if(istype(L))
+		if(prob(25))
+			L.Weaken(3)
+			L.visible_message("<span class='danger'>\The [src] knocks down \the [L]!</span>")
```
Maneater-а на локалке кто-нибудь проверял как работает?

TobyThorne replied:
Сто тыщ раз проверил, пока делал ПР, ибо сделать ему приручение оказалось задачей довольно таки нетривиальной. В целом, положение такое:
1. Робастный, шо пиздец, один на один без огнестрела или брони победить - это надо быть Соляром. Хотя и слабее, чем на Гунах.
2. Приручение работает надёжно как швейцарские часы. Можно успешно делать ферму ксеномяса. Всех остальных, за исключением других мобов из фракции "creatures" (томаты-убийцы и не-культовые зелёные кусаки) идёт разбирать на атомы. 
3. Первые пять секунд жизни, во время которых можно приручить, стоит и аутирует.
4. Если не собрать вручную с грядки - соберётся через какое-то время сам.
5. Ghost Possess работает.

TobyThorne replied:
Ну и кормление бесполезно, если его поссессят, да. Управляющий им игрок может грызть, кого хочет. Это уже особо никак не исправить. Разве что впилить ему интерфейс, показывающий друга, но это уже нужно много кода.

Epicus7 replied:
Оппа, а может убрать ghost-possess, если приручили? А то неоч получается, ты приручал-приручал, а тут госты приходят и все обламывают.

Epicus7 replied:
И первые пять секунд тоже, чтобы нельзя было перехватить до приручения.

TobyThorne replied:
Сделяль.

Epicus7 commented:
Чот у меня сомнения есть, тестировал ли это нормально автор и насколько это тестировал Тапок. Если Тапок нормально протестировал (явно напишет), то ок.

PlinHost commented on code/game/objects/items/toys.dm:
```diff
@@ -937,3 +938,21 @@
 				H.drop_from_inventory(W)
 		playsound(user.loc, pick('sound/effects/xylophone1.ogg','sound/effects/xylophone2.ogg','sound/effects/xylophone3.ogg'), 60)
+
+/obj/item/toy/banbanana
+	name = "BANana"
+	desc = "What happens if I peel it?"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "banana"
+
+/obj/item/toy/banbanana/attack_self(mob/user)
+	for(var/mob/M in viewers(user, null))
+		if (M.client)
+			M.show_message("<span class='danger'>You have been banned by HO$T.\nReason: Honk.</span>")
+			M.show_message("<span class='warning'>This is a PERMENANT ban.</span>")
+			if(ishuman(M))
+				var/mob/living/carbon/human/H = M
+				H.eye_blind += 1
+	playsound(user.loc, 'sound/effects/adminhelp.ogg', 100)
+	user.drop_from_inventory(src)
+	qdel(src)
```
> Почему всем игрокам пишется, что они "banned", но звук только у того проигрывается, кто активировал?

Звук проигрывается на турфе с дефолтным радиусом, что не так?

Epicus7 replied:
А ну да, тогда со звуком все норм. Но остальные вопросы остаются.

TobyThorne replied:
>Кажется тут явно не сердечный приступ, а просто слепота? Если оно вообще работает.

У персонажей - секундная слепота, у игроков - сердечный приступ. :smekalo4ka:

>Может быть это и смешно, но если будут часто абузить - хзхз.

Ну-с, его ещё нужно получить через мутации, растёт он медленно и плодоносит одной штукой за раз. Мутаген имеет шанс вовсе убить yield. Да и размножить их через плоды не получится, только отрезать сэмплы. И фича, в общем-то, не настолько смешная чтобы прям дико абузить.

TobyThorne commented on code/modules/hydroponics/grown.dm:
```diff
@@ -188,7 +188,7 @@
 				qdel(W)
 				return
-		else if(W.sharp)
+		else if(W.sharp && !istype(W,/obj/item/weapon/reagent_containers/syringe))
```
Ну, тут проблема не столько в нарезании шприцом (шприцем?), сколько в невозможности вколоть им реагенты в нарезаемые фрукты. Нарезание же всякими отвёртками - скорее фича, чем баг.

Epicus7 replied:
Чот такая себе фича. Давай вообще проверку уберем и будем jackboots-ами нарезать, еще смешнее будет

TobyThorne replied:
Ну хз-хз. Могу изменить, в общем-то. Хозяин - барин.

Epicus7 replied:
Да я не геймдиз, а им пофиг. Смотри сам как хочешь.

TobyThorne committed:
Update grown.dm

Теперь фрукты-овощи нарезаются только предметами с лезвием - ножи, топорики, etc.

TobyThorne committed:
Update maneater.dm

Теперь накормленного Maneater-а нельзя поссесснуть.

TapKo4eB commented on code/modules/mob/living/simple_animal/hostile/maneater.dm:
```diff
@@ -0,0 +1,71 @@
+
+	var/angry = 0
+
+/mob/living/simple_animal/hostile/maneater/Initialize()
+	. = ..()
+
+	if(name == initial(name))
+		name = "[name] ([sequential_id(/mob/living/simple_animal/hostile/maneater)])"
+	real_name = name
+
+	spawn(5 SECONDS)
+		if(!angry)
+			controllable = TRUE
```
чтобы ты понимал на стадии /mob/living/Initialize() сканируется на контроллабл и добавляется в список поссеса. Своим изменением контроллабла после вызова родича ты сломал весь поссес))0

TobyThorne replied:
У блят. Ладно, чёнить придумаю.

TobyThorne committed:
Update maneater.dm

Maneater is no longer Ghost-possessible

TobyThorne commented:
Убрал возможность поссессить людоедов. В общем-то, раз уж их можно приручать, то давать их игрокам - перебор. Да и слишком робастные. Если под контролем более-менее рукастого космонавтика будет стрейфить и доджить выстрелы - так можно пол-станции вырезать.

ArmenianOnyx commented:
Мне кажется людоедов лучше оставить. Робастных мобов на станции нема.

TapKo4eB commented:
> 
> 
> Мне кажется людоедов лучше оставить. Робастных мобов на станции нема.

а кто сказал что их удалили

ArmenianOnyx commented:
> Убрал возможность поссессить людоедов. В общем-то, раз уж их можно приручать, то давать их игрокам - перебор. Да и слишком робастные. Если под контролем более-менее рукастого космонавтика будет стрейфить и доджить выстрелы - так можно пол-станции вырезать.

А это шо?

ArmenianOnyx commented:
Бля, неправильно прочитал. Посеять и поссесить

TapKo4eB reviewed (approved):
короче прикольно
